src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/CcdUpdater.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.events;
+
+import com.google.common.base.Strings;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CaseRetriever;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+@Service
+public class CcdUpdater {
+
+    private final AttachDocsToSupplementaryEvidence attachDocsPublisher;
+    private final CreateExceptionRecord exceptionRecordCreator;
+    private final CaseRetriever caseRetriever;
+
+    public CcdUpdater(
```

Comment:
This name needs changing CaseHandler or something similar might be better because this class handles update as well as create

Reply:
After quick discussion we agreed on `EnvelopeHandler`